homeassistant/components/blebox/binary_sensor.py

```diff
@@ -0,0 +1,62 @@
+"""BleBox sensor entities."""
+from dataclasses import dataclass
+
+from blebox_uniapi.binary_sensor import BinarySensor as BinarySensorFeature
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import BleBoxEntity, create_blebox_entities
+
+
+@dataclass
+class BleboxBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """Class describing Blebox binary sensor entities."""
+
+
+BINARY_SENSOR_TYPES = (
+    BleboxBinarySensorEntityDescription(
+        key="moisture",
+        device_class=BinarySensorDeviceClass.MOISTURE,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a BleBox entry."""
+
+    create_blebox_entities(
+        hass,
+        config_entry,
+        async_add_entities,
+        BleBoxBinarySensorEntity,
+        "binary_sensors",
+    )
+
+
+class BleBoxBinarySensorEntity(BleBoxEntity, BinarySensorEntity):
+    """Representation of a BleBox binary sensor feature."""
+
+    def __init__(self, feature: BinarySensorFeature) -> None:
+        """Initialize a BleBox sensor feature."""
+        super().__init__(feature)
+
+        for description in BINARY_SENSOR_TYPES:
+            if description.key == feature.device_class:
+                self.entity_description = description
+                break
```

Comment:
This is bad behavior.
It shouldn't be part of `__init__` and the logic should be moved to `async_setup_entry`

Reply:
@epenet Can you elaborate? As I see in major other integrations, self.entity_description is set on the entity's init. I would like to keep it consistent. 
Your suggestion would only resolve descriptions for `BleBoxBinarySensorEntity`, but there is `BleBoxSensorEntity` which also uses entity_descriptors. Also visible prospect application of entity_descriptort in other blebox platforms f.e. light platform.

Reply:
self.entity_description is always set on the entity's init.
BUT it is normally passed as an argument, not looked up during the entity's init.
The lookup should be done BEFORE the entity's init.

Reply:
Regarding `BleBoxSensorEntity`, in a follow-up PR I suggest that apply the same logic that I recommended here and remove reference to the central `create_blebox_entities`.
I just don't feel that `create_blebox_entities` method adds much value, and just makes it harder to review the code.

Or maybe add a new helper method so that you share one loop between all platforms, but keep the class instance creating inside each platform:
```
@callback
def get_blebox_features(hass, config_entry, entity_type) -> list[Feature]`:
```